upgrade_tests/helpers/existence.py

```diff
@@ -103,6 +104,60 @@
 attribute_keys['content-view'] = 'content view id'
 
 
+def _find_on_list_of_dicts(lst, data_key, all_=False):
+    """Returns the value of a particular key in a dictionary from the list of
+    dictionaries, when 'all' is set to false.
+
+    When 'all' is set to true, returns the list of values of given key from all
+    the dictionaries in list.
+
+    :param list lst: A list of dictionaries
+    :param str data_key: A key name of which data to be retrieved from given
+        list of dictionaries
+    :param bool all: Fetches all the values of key in list of dictionaries if
+        True, else Fetches only single and first value of a key in list of
+        dictionaries
+    :returns the list of values or a value of a given data_key depends on
+        'all' parameter
+
+    """
+    dct_values = [dct.get(data_key) for dct in lst]
+    if all_:
+        return dct_values
+    for v in dct_values:
+        if v is not None:
+            return v
+
+    raise KeyError(
+        'Unable to find data for key \'{0}\' in satellite.'.format(
+            data_key))
+
+
+def _find_on_list_of_dicts_using_search_key(lst_of_dct, search_key, attr):
+    """Returns the value of attr key in a dictionary from the list of
+    dictionaries with the help of search_key.
+
+    To retrieve the value search key and the attribute should be in the
+    same dictionary
+
+    :param list lst: A list of dictionaries
+    :param str search_key: A value of any unique key in dictionary in list of
+        dictionary.
+        The value will be used to fetch another keys value from same dictionary
+    :param str attr: The key name in dictionary in which search_key exists in
```

Comment:
I am still confuse don why the `key` is called `attr` here?

In python `attr` is object attributes accessed with `.` e.g: `object.attribute` and `key` are dictionary items `object['key']`  not better to call it `key` to avoid confusion ?